src/ipa_pc/mod.rs

```diff
@@ -111,21 +111,43 @@ impl<G: AffineCurve, D: Digest> InnerProductArgPC<G, D> {
         for (labeled_commitment, value) in labeled_commitments.zip(values) {
             let commitment = labeled_commitment.commitment();
             combined_v += &(cur_challenge * &value);
-            combined_commitment_proj += &labeled_commitment.commitment().comm.mul(cur_challenge);
+
+            if commitment.comm.len() > 1 {
+
+                let mut comm_lc = <G::Projective as ProjectiveCurve>::zero();
+                for (i, comm_single) in commitment.comm.iter().enumerate() {
+                    let is = i * n;
+                    comm_lc += &comm_single.mul(point.pow(&[is as u64]));
+                }
+                combined_commitment_proj += &comm_lc.mul(&cur_challenge);
+
+            } else {
+
+                combined_commitment_proj += &commitment.comm[0].mul(cur_challenge);
+            }
+
             cur_challenge = opening_challenges(opening_challenge_counter);
             opening_challenge_counter += 1;
 
             let degree_bound = labeled_commitment.degree_bound();
-            assert_eq!(degree_bound.is_some(), commitment.shifted_comm.is_some());
 
-            if let Some(degree_bound) = degree_bound {
-                let shift = point.pow([(vk.supported_degree() - degree_bound) as u64]);
+            let d = degree_bound.and_then(|d| {
```

Comment:
We double use d in the succinct_check(). I would propose for a different naming here.

Reply:
Done